applications/services/storage/storage_processing.c

```diff
@@ -513,46 +513,26 @@ static FS_Error storage_process_sd_status(Storage* app) {
 
 /******************** Aliases processing *******************/
 
-void storage_process_alias(
-    Storage* app,
-    FuriString* path,
-    FuriThreadId thread_id,
-    bool create_folders) {
-    if(furi_string_start_with(path, STORAGE_APP_DATA_PATH_PREFIX)) {
-        FuriString* apps_data_path_with_appsid = furi_string_alloc_set(APPS_DATA_PATH "/");
-        furi_string_cat(apps_data_path_with_appsid, furi_thread_get_appid(thread_id));
-
-        // "/data" -> "/ext/apps_data/appsid"
-        furi_string_replace_at(
-            path,
-            0,
-            strlen(STORAGE_APP_DATA_PATH_PREFIX),
-            furi_string_get_cstr(apps_data_path_with_appsid));
+void storage_process_alias(Storage* app, FuriString* path, bool create_folders) {
+    if(furi_string_start_with(path, STORAGE_APPS_DATA_STEM "/")) {
+        FuriString* apps_data_appid = furi_string_alloc_set(path);
+        // Only create "/ext/apps_data/appid", not any subdir
+        // "/ext/apps_data/appid/whatever" -> "/ext/apps_data/appid"
+        size_t slash =
+            furi_string_search_char(apps_data_appid, '/', strlen(STORAGE_APPS_DATA_STEM "/"));
+        if(slash != FURI_STRING_FAILURE) {
+            furi_string_left(apps_data_appid, slash);
+        }
 
         // Create app data folder if not exists
-        if(create_folders &&
-           storage_process_common_stat(app, apps_data_path_with_appsid, NULL) != FSE_OK) {
-            furi_string_set(apps_data_path_with_appsid, APPS_DATA_PATH);
-            storage_process_common_mkdir(app, apps_data_path_with_appsid);
-            furi_string_cat(apps_data_path_with_appsid, "/");
-            furi_string_cat(apps_data_path_with_appsid, furi_thread_get_appid(thread_id));
-            storage_process_common_mkdir(app, apps_data_path_with_appsid);
+        if(create_folders && storage_process_common_stat(app, apps_data_appid, NULL) != FSE_OK) {
```

Comment:
This code incorrectly handles paths like `"/ext/apps_data/nfc"`
See https://github.com/flipperdevices/flipperzero-firmware/actions/runs/10283770489/job/28458336904#step:7:89

@Willy-JL @DrZlo13

Reply:
seems like its still the error i mentioned above, setting up the CI unit tests, not running them. code that causes the issue is not in flipper code, but python helper is failing when creating a folder that already exists. i had tried in 887cd4b94b84e61587cbfb0ccab9968807930927 to make it ignore an "already exists" error but seems like this fix had some syntax issue.

i will fix this, but maybe could consider making it so checking for dir exist in "/ext/apps_data/*" will say that dir already exists?

Reply:
I think maintaining compatibility and existing behaviour from API is quite important - looking forward to hearing @DrZlo13's opinion on these changes.